### ISSUE TITLE
`TIMESTAMP(p)` datatype not parsed as single keyword in MySQL dialect

### ISSUE THREAD

**Issue by stonio** [sqlfluff/sqlfluff#6846]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Same issue as #5589 for MySQL.

> MySQL has fractional seconds support for [TIME](https://dev.mysql.com/doc/refman/8.4/en/time.html), [DATETIME](https://dev.mysql.com/doc/refman/8.4/en/datetime.html), and [TIMESTAMP](https://dev.mysql.com/doc/refman/8.4/en/datetime.html) values, with up to microseconds (6 digits) precision.

Source: https://dev.mysql.com/doc/refman/8.4/en/fractional-seconds.html

### Expected Behaviour

SQLFluff should recognize `TIMESTAMP(p)` (and variants)  as a single keyword token.
MySQL supports resolutions (p) from 0 (no fractional seconds) to 6 (microseconds ).

### Observed Behaviour

Error LT01 is raised : suggest to add single whitespace between keyword and start bracket `(`:
- `TIME (2)`
- `DATETIME (2)`
- `TIMESTAMP (2) `
- `CURRENT_TIMESTAMP (6)`

### How to reproduce

- Example from https://dev.mysql.com/doc/refman/8.4/en/fractional-seconds.html
    ```sql
    CREATE TABLE fractest(c1 TIME(2), c2 DATETIME(2), c3 TIMESTAMP(2));
    ```
- Example from https://dev.mysql.com/doc/refman/8.4/en/timestamp-initialization.html
    ```sql
    CREATE TABLE t1 (
        ts TIMESTAMP(6) DEFAULT CURRENT_TIMESTAMP (6) ON UPDATE CURRENT_TIMESTAMP(6)
    );
    ```

### Dialect

MySQL

### Version

3.4.0

### Configuration

No configuration file was used. This can be triggered with the default configuration or via <https://online.sqlfluff.com/fluffed?sql=&dialect=MySQL>.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
